tests/unit/test_schemas.py

```diff
@@ -1,390 +0,0 @@
-        for cls in protocol.getProtocolClasses():
-            instance = factory(cls)
-            jsonDict = instance.toJsonDict()
-            self.assertTrue(cls.validate(jsonDict))
-
-    def testValidateDefaultValues(self):
-        self.validateClasses(self.getDefaultInstance)
-
-    def testValidateTypicalValues(self):
-        self.validateClasses(self.getTypicalInstance)
-
-    def testValidateRandomValues(self):
-        self.validateClasses(self.getRandomInstance)
-
-    def testValidateBadValues(self):
-        for cls in protocol.getProtocolClasses():
-            instance = self.getTypicalInstance(cls)
-            jsonDict = instance.toJsonDict()
-            self.assertFalse(cls.validate(None))
-            self.assertFalse(cls.validate([]))
-            self.assertFalse(cls.validate(1))
-            # setting values to bad values should be invalid
-            for key in jsonDict.keys():
-                dct = dict(jsonDict)
-                dct[key] = self.getInvalidValue(cls, key)
-                self.assertFalse(cls.validate(dct))
-            for c in utils.powerset(jsonDict.keys(), 10):
-                if len(c) > 0:
-                    dct = dict(jsonDict)
-                    for f in c:
-                        dct[f] = self.getInvalidValue(cls, f)
-                    self.assertFalse(cls.validate(dct))
-
-
-class GetProtocolClassesTest(SchemaTest):
-    """
-    Tests the protocol.getProtocolClasses() function to ensure it
-    works correctly.
-    """
-    def testAllClasses(self):
-        classes = protocol.getProtocolClasses()
-        self.assertGreater(len(classes), 0)
-        for class_ in classes:
-            self.assertTrue(issubclass(class_, protocol.ProtocolElement))
-
-    def testRequestAndResponseClasses(self):
-        requestClasses = protocol.getProtocolClasses(protocol.SearchRequest)
-        responseClasses = protocol.getProtocolClasses(protocol.SearchResponse)
-        self.assertEqual(len(requestClasses), len(responseClasses))
-        self.assertGreater(len(requestClasses), 0)
-        for class_ in requestClasses:
-            self.assertTrue(issubclass(class_, protocol.SearchRequest))
-        for class_ in responseClasses:
-            self.assertTrue(issubclass(class_, protocol.SearchResponse))
-            valueListName = class_.getValueListName()
-            self.assertGreater(len(valueListName), 0)
-
-
-class EmbeddedValuesTest(SchemaTest):
-    """
-    Tests for the embedded values maps in ProtocolElements.
-    """
-    def testEmbeddedValues(self):
-        for cls in protocol.getProtocolClasses():
-            for member in cls.__slots__:
-                if cls.isEmbeddedType(member):
-                    instance = cls.getEmbeddedType(member)()
-                    self.assertIsInstance(instance, protocol.ProtocolElement)
-                else:
-                    self.assertRaises(KeyError, cls.getEmbeddedType, member)
-
-
-class SearchResponseBuilderTest(SchemaTest):
```

Comment:
Yes, these are pretty good tests actually, we should keep them. I think they can be ported fairly easily by replacing getTypicalInstance with by just instantiating the class_(), and getting rid of ``getRandomInstance()``. So, whereever you see ``getTypicalInstance(class_)`` replace it with ``class_()``, and delete any getRandomInstance you see. That should make ``SearchResponseBuilderTest`` fairly portable.